cmd/rookd/daemon.go

```diff
@@ -47,7 +47,7 @@ func runDaemon(cmd *cobra.Command, args []string) error {
 	if err := flags.VerifyRequiredFlags(daemonCmd, []string{"type"}); err != nil {
 		return err
 	}
-	if daemonType != "mon" && daemonType != "osd" && daemonType != "mds" && daemonType != "rgw" {
+	if daemonType != "mon" && daemonType != "osd" && daemonType != "mds" && daemonType != "rgw" && daemonType != "rgw-admin" {
 		return fmt.Errorf("unknown daemon type: %s", daemonType)
```

Comment:
Agreed, daemon doesn't apply to all of them, but i think they're still in the same category. What about the following?
- rookd has two commands: `daemon` and `tool`. This makes it clear to the caller what type of thing it is running.
- internally, the call to `cephd.New().RunDaemon()` is renamed to `cephd.New().Run()` since it doesn't really care if it's running a daemon or a tool, it's just calling into libcephd.